app/src/main/java/com/gsandchase/mileagelogger/MainActivity.java

```diff
@@ -50,24 +47,18 @@ protected void onCreate(Bundle savedInstanceState) {
         addMileageButton = (Button) findViewById(R.id.addMileageButton);
         newMileage = (EditText) findViewById(R.id.newMileage);
 
+        // TODO: Read in saved mileage data
+        mileageCounter = new MileageCounter();
 
-        readMileage();
-
-        if(newWeek(DATE)) {
-            weeklyMileage = 0;
-            weeklyMileageS = "0";
-        }
 
-        totalMileage.setText(weeklyMileageS);
+        updateCounterText();
 
 
         addMileageButton.setOnClickListener(new View.OnClickListener() {
             public void onClick(View v) {
```

Comment:
Changed the onClick handler to do two things: update the mileage counter internally, and trigger an update of the on-screen counter.